### PR TITLE
New table view react

### PR DESCRIPTION
**Pull request by thealmightyone**
Basic functionality in React

### CODE REVIEW

File: app/js/views/appView.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import SchemaCollection from './../models/schemaCollection';
 
 import SidebarView from './sidebarView';
 import HeaderView from './headerView';
-import TableView from './tableView';
+import TableView from './newTableView';
```
Comment (stepek) [inline app/js/views/appView.js]: In this commit you imported not existing module.

Comment (stepek) [inline app/js/views/appView.js]: Fixed

**Comment by stepek** [comment]
Please add ```react-bootstrap``` to ```package.json```

File: app/js/views/components/tableRow.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import EditButton from './editButton.jsx';
+
+export default class TableRow extends React.Component {
+  constructor(params) {
+    super(params);
+  }
+
+  render() {
+    const item = this.props.item;
+    let list = this.props.schema.schema.propertiesOrder.map((key, index) => {
```
Comment (stepek) [inline app/js/views/components/tableRow.jsx]: Should be const.

File: app/js/views/components/tableRow.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      const property = this.props.schema.schema.properties[key];
+      const view = property['view'];
+
+      if (view && !view.includes('list')) {
+        return null;
```
Comment (stepek) [inline app/js/views/components/tableRow.jsx]: Why you return null not undefined?

File: app/js/views/components/tableRow.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      }
+      const title = property.title.toLowerCase();
+
+      var urlPattern = /^((http|https):\/\/)/;
+
+      try {
+        if (property.type === 'object' || property.originalType === 'object') {
+          if (item[key] === undefined) {
+            return null;
```
Comment (stepek) [inline app/js/views/components/tableRow.jsx]: Why you return null not undefined?

File: app/js/views/components/tableRow.jsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    let list = this.props.schema.schema.propertiesOrder.map((key, index) => {
+      const property = this.props.schema.schema.properties[key];
+      const view = property['view'];
+
+      if (view && !view.includes('list')) {
+        return null;
+      }
+      const title = property.title.toLowerCase();
+
+      var urlPattern = /^((http|https):\/\/)/;
```
Comment (stepek) [inline app/js/views/components/tableRow.jsx]: Please do not use vars.

File: app/js/views/components/tableRow.jsx (hunk 4 above)
Comment (thealmightyone) [inline app/js/views/components/tableRow.jsx]: false, null and undefined generate no element in React,
whereas  "", 0 and NaN generate <span> elements

Comment (stepek) [inline app/js/views/components/tableRow.jsx]: so you can change ```return null``` to ```return```

**Comment by nati** [comment]
I don't get what we got by applying react..
For me, existing code looks like more simple to understand & easy to extend..

**Comment by thayashi** [comment]
So, Gohan web ui is still kinda prototyping phase, I understand keeping current way is much convenient for  prototyping. But eventually I'd like to have react way in the near future. let's keep this repo for the feature.

**Pull request closed without merging**